### ISSUE TITLE
Error when running virsorter=2.2.3

### ISSUE THREAD

**Issue by jq1242560952** [jiarong/VirSorter2#110]
Dear jiarong
When I running virsorter, there is no error on the local laptop, but on the server, it keep getting an error when I run "virsorter run -w test9.out -i 1.fna --min-length 1500 -j 4 all", the error as shown below:

[2022-07-06 09:49 ERROR] See error details in /home/zhzhu/jq/software/test9.out/log/iter-0/step2-extract-feature/extract-feature-from-gff-common.log
[Wed Jul  6 09:49:50 2022]
Error in rule gff_feature:
    jobid: 15
    output: iter-0/all.pdg.gff.ftr
    log: log/iter-0/step2-extract-feature/extract-feature-from-gff-common.log (check log file(s) for error message)
    conda-env: /home/zhzhu/jq/software/db/conda_envs/88290cce
    shell:

        python /home/zhzhu/miniconda3/envs/vs2/lib/python3.9/site-packages/virsorter/./scripts/extract-feature-from-gff.py /home/zhzhu/jq/software/db/rbs/rbs-catetory.tsv iter-0/all.pdg.gff iter-0/all.pdg.gff.ftr &> log/iter-0/step2-extract-feature/extract-feature-from-gff-common.log || { echo "See error details in /home/zhzhu/jq/software/test9.out/log/iter-0/step2-extract-feature/extract-feature-from-gff-common.log" | python /home/zhzhu/miniconda3/envs/vs2/lib/python3.9/site-packages/virsorter/./scripts/echo.py --level error; exit 1; }

        (one of the commands exited with non-zero exit code; note that snakemake uses bash strict mode!)

Exiting because a job execution failed. Look above for error message**

I checked th log "extract-feature-from-gff-common.log", and the content is
 

"Traceback (most recent call last):
  File "/home/zhzhu/miniconda3/envs/vs2/lib/python3.9/site-packages/virsorter/./scripts/extract-feature-from-gff.py", line 17, in <module>
    from virsorter.utils import (
  File "/home/zhzhu/anaconda3/lib/python3.8/site-packages/virsorter-2.2.3-py3.8.egg/virsorter/utils.py", line 15, in <module>
    DEFAULT_CONFIG = get_default_config()
  File "/home/zhzhu/anaconda3/lib/python3.8/site-packages/virsorter-2.2.3-py3.8.egg/virsorter/config.py", line 26, in get_default_config
    assert os.path.isfile(TEMPLATE)
AssertionError
"

I tried many ways to fix this, including updating the python version to the same as my laptop (3.9.12), reinstalling virsorter, but there is same error.

Thanks!

**Comment by jiarong** [comment]
Looks like you did not run the database setup step (`virsorter setup -d <db>`)? Every time VirSorter2 is reinstalled, this step needs to be re-run.
Or you can use installation option 3 if you have singularity on your server.

**Comment by jq1242560952** [comment]
Thank for your quick response. Yes, I re-run the setup step when the Virsorter is reinstalled, but there is  same error. And how do I know if I have singularity in my server, and when I run the virsorter2.sif, can I skip the download db step, and use the absolute path of this file to replace virsorter in commands, for example "/home/software/virsorter2.sif run -w test.out -i test.fa --min-length 1500 -j 4 all"?

**Comment by jiarong** [comment]
when you re-run the setup step, make sure to delete the previous database first.
type in `singularity` in command line. if help message show up, then you have it. The command you described is correct.

**Comment by jq1242560952** [comment]
Thanks for your suggestion, we submit to the server by Portable Batch System (PBS), there is the content of pbs:

#!/usr/bin/bash

#PBS -l nodes=1:ppn=2

#PBS -N virsorter

#PBS -q cu

cd /home/zhzhu/jq/

conda activate vs2

/home/zhzhu/jq/virsorter2.sif run -w prophage.out -i /home/zhzhu/jq/total.fna --min-length 1500 -j 2 all

but there is an error:
/usr/bin/env: run-singularity: No such file or directory

Can't find file /home/zhzhu/.vnc/mu01:43.pid
You'll have to kill the Xvnc process manually

Then I want to know if the server have singularity, I adevised the content of pbs:
#!/usr/bin/bash

#PBS -l nodes=1:ppn=2

#PBS -N virsorter

#PBS -q cu

cd /home/zhzhu/jq/pro/

singularity build virsorter2.sif docker://jiarong/virsorter:latest.

There is an error:
/opt/tsce4/torque6/share/cu07/mom_priv/jobs/127060.mu01.SC: line 11: singularity: command not found

Can't find file /home/zhzhu/.vnc/mu01:44.pid
You'll have to kill the Xvnc process manually.

**Does this mean I don't have singularity on my server?**

**Comment by jq1242560952** [comment]
I solved the above problem, and got there files when run 1 fasta file on the server. And I want to run a lot of fasta of bacteria (abot 4,000 bacteria genome), so **I merged all the fasta files into one file**, named total.fasta, and input same command as 1 fasta file. But **no three result files appear when running for three day**s, is this because I shouldn't combine them into one file？If is, so how should I analyze the genomes of more than 4000 bacterial strains?
